File: spl-net/src/main/java/bgu/spl/net/impl/messages/ErrorMessage.java
package bgu.spl.net.impl.messages;

public class ErrorMessage extends Message{
    private short messageOpCode;
    public ErrorMessage(short _messageOpCode) {
        super((short)11);
        messageOpCode = _messageOpCode;
    }

    public short getMessageOpCode(){return messageOpCode;}

    public String toString(){return "11"+messageOpCode;}
}

File: Boost_Echo_Client/src/ErrorMessage.cpp
//
// Created by MAY on 04/01/2022.
//

#include "../include/ErrorMessage.h"

ErrorMessage::ErrorMessage(short _messageOpcode, short _opCode) : Message(_opCode), messageOpcode(_messageOpcode) {

}

short ErrorMessage::getMessageopcode() {
    return messageOpcode;
}

std::string ErrorMessage::toString() {
    std::string answer = "ERROR ";
    answer.append(std::to_string(getMessageopcode()));
    return answer;
}
